pkg/monitortestlibrary/pathologicaleventlibrary/duplicated_event_patterns.go

```diff
@@ -2,419 +2,569 @@ package pathologicaleventlibrary
+	Name:              "AllowImagePullBackOffFromRedHatRegistry",
+	MessageHumanRegex: regexp.MustCompile(`Back-off pulling image .*registry.redhat.io`),
 }
 
-func TopologyPointer(topology v1.TopologyMode) *v1.TopologyMode {
-	return &topology
+// Separated out in testBackoffStartingFailedContainer
+var AllowBackOffRestartingFailedContainer = &AllowedDupeEvent{
+	Name:               "AllowBackOffRestartingFailedContainer",
+	MessageReasonRegex: regexp.MustCompile(`^BackOff$`),
+	MessageHumanRegex:  regexp.MustCompile(`Back-off restarting failed container`),
 }
 
-func PlatformPointer(platform v1.PlatformType) *v1.PlatformType {
-	return &platform
+// Separated out in testRequiredInstallerResourcesMissing
+var EtcdRequiredResourcesMissing = &AllowedDupeEvent{
+	Name:               "EtcdRequiredResourcesMissing",
+	MessageReasonRegex: regexp.MustCompile(`^RequiredInstallerResourcesMissing$`),
+	MessageHumanRegex:  regexp.MustCompile(`secrets: etcd-all-certs-[0-9]+`),
 }
 
-func StringPointer(testSuite string) *string {
-	return &testSuite
+// reason/OperatorStatusChanged Status for clusteroperator/etcd changed: Degraded message changed from "NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: 2 of 3 members are available, ip-10-0-217-93.us-west-1.compute.internal is unhealthy" to "NodeControllerDegraded: All master nodes are ready\nEtcdMembersDegraded: No unhealthy members found"
+var EtcdClusterOperatorStatusChanged = &AllowedDupeEvent{
+	Name: "EtcdClusterOperatorStatusChanged",
+	LocatorKeyRegexes: map[monitorapi.LocatorKey]*regexp.Regexp{
+		monitorapi.LocatorNamespaceKey: regexp.MustCompile(`openshift-etcd`),
+		monitorapi.LocatorPodKey:       regexp.MustCompile(`^openshift-etcd`),
+	},
+	MessageReasonRegex: regexp.MustCompile(`^OperatorStatusChanged$`),
+	MessageHumanRegex:  regexp.MustCompile(`Status for clusteroperator/etcd changed.*No unhealthy members found`),
 }
 
-// isConsoleReadinessDuringInstallation returns true if the event is for console readiness and it happens during the
-// initial installation of the cluster.
-// we're looking for something like
-// > ns/openshift-console pod/console-7c6f797fd9-5m94j node/ip-10-0-158-106.us-west-2.compute.internal - reason/ProbeError Readiness probe error: Get "https://10.129.0.49:8443/health": dial tcp 10.129.0.49:8443: connect: connection refused
-// with a firstTimestamp before the cluster completed the initial installation
-func isConsoleReadinessDuringInstallation(monitorEvent monitorapi.Interval, kubeClientConfig *rest.Config, _ int) (bool, error) {
-	if !strings.Contains(monitorEvent.Locator, "ns/openshift-console") {
-		return false, nil
-	}
-	if !strings.Contains(monitorEvent.Locator, "pod/console-") {
-		return false, nil
-	}
-	if !strings.Contains(monitorEvent.Locator, "Readiness probe") {
-		return false, nil
-	}
-	if !strings.Contains(monitorEvent.Locator, "connect: connection refused") {
-		return false, nil
```

Comment:
This was broken, we're checking for things in the locator that would never be in the locator, they're from the message. If it's been in this busted for ages, we don't need it. Removed.